recipe/meta.yaml

```
@@ -1,34 +1,117 @@
+{% set ignore_modules = ignore_modules + " --ignore=test/contrib/einstein/test_stein_kernels.py" %}
+{% set ignore_modules = ignore_modules + " --ignore=test/contrib/einstein/test_steinvi.py" %}
+{% set ignore_modules = ignore_modules + " --ignore=test/contrib/einstein/test_stein_loss.py" %}
+{% set ignore_modules = ignore_modules + " --ignore=test/contrib/einstein/test_steinvi_util.py" %}
+{% set ignore_modules = ignore_modules + " --ignore=test/contrib/test_module.py" %}
+{% set ignore_modules = ignore_modules + " --ignore=test/contrib/test_nested_sampling.py" %}
+{% set ignore_modules = ignore_modules + " --ignore=test/contrib/test_tfp.py" %}
+{% set ignore_modules = ignore_modules + " --ignore=test/contrib/hsgp/test_approximation.py" %}
+{% set ignore_modules = ignore_modules + " --ignore=test/contrib/hsgp/test_spectral_densities.py" %}
+{% set ignore_modules = ignore_modules + " --ignore=test/contrib/stochastic_support/test_sdvi.py" %}
+
+# test/infer directory missing::
+# funsor >=0.4.1(all_platforms)
+# optax(win and osx)
+{% set ignore_modules = ignore_modules + " --ignore=test/infer/test_gradient.py" %}
+{% set ignore_modules = ignore_modules + " --ignore=test/infer/test_autoguide.py" %}
+{% set ignore_modules = ignore_modules + " --ignore=test/infer/test_hmc_gibbs.py" %}
+{% set ignore_modules = ignore_modules + " --ignore=test/infer/test_hmc_gibbs.py" %}
+{% set ignore_modules = ignore_modules + " --ignore=test/infer/test_infer_util.py" %}
+{% set ignore_modules = ignore_modules + " --ignore=test/infer/test_mcmc.py" %}
+{% set ignore_modules = ignore_modules + " --ignore=test/infer/test_svi.py" %}
+
+# test/pyroapi directory missing::
+# pyro-api >=0.1.1
+{% set ignore_modules = ignore_modules + " --ignore=test/pyroapi/test_pyroapi.py" %}
+
+# test directory missing:
+# funsor >=0.4.1
+{% set ignore_modules = ignore_modules + " --ignore=test/test_distributions.py" %}
+# tensorflow_probability >=0.18.0 (Only 1 test)
+{% set deselect_tests =                  " --deselect=test/test_distributions_util.py::test_no_tracer_leak_at_lazy_property_sample" %}
+# funsor >=0.4.1
+{% set ignore_modules = ignore_modules + " --ignore=test/test_examples.py" %}
+# graphviz
+{% set deselect_tests = deselect_tests + " --deselect=test/test_model_rendering.py::test_render_model_filename" %}
+# chex==0.1.84(all_platforms) and optax(win and osx)
+{% set ignore_modules = ignore_modules + " --ignore=test/test_optimizers.py" %}
+# funsor >=0.4.1 missing
+{% set ignore_modules = ignore_modules + " --ignore=test/test_pickle.py" %}
+
 test:
+  source_files:
+    - test
+    - examples
   imports:
     - numpyro
+  commands:
+    - pip check
+    - pytest -v {{ ignore_modules }} {{ deselect_tests }} test
+  requires:
+    - pip
+    - pytest
+    # optax is not avaliable on win-64 and osx-64 
+    # osx-arm64 consist optax but don`t have openssl[version='>=1.1.1j,<1.1.2a as dep by optax
+    - optax>=0.0.6 # [not (win or osx)]
+    - scikit-learn
+    - matplotlib
```

```suggestion
    - matplotlib-base
```
A lightweight variant without `pyqt`